cookbooks/aws-parallelcluster-environment/templates/cfn_bootstrap/cfn-hup.conf.erb

```diff
@@ -0,0 +1,5 @@
+[main]
+stack=<%= @stack_id %>
+region=<%= @region %>
+url=<%= @cloudformation_url %>
+role=<%= @cfn_init_role %>
```

Comment:
Why adding this new parameter "role"?

Reply:
Very good question :-)= 
This is required by cfn-hup when you configure it to monitor a nested stack. 
Compute fleet and login nodes are defined in nested stacks so it's required for me.
It is optional for the head node, instead, but we prefer to have a single definition that works for all.

We added this role as part of the DFSMv2 feature, but we are not introducing it in this PR specifically. In fact, it was already defined [here](https://github.com/aws/aws-parallelcluster/blob/develop/cli/src/pcluster/resources/compute_node/user_data.sh#L62)